ENIGMAsystem/SHELL/Widget_Systems/General/tinyfiledialogs/dialogs.cpp

```diff
@@ -403,6 +438,7 @@ namespace enigma_user
     fname_or_dir = tfd_add_escaping(fname_or_dir);
     titlebar = tfd_add_escaping(titlebar);
     filter = tfd_add_escaping(filter);
+    filter = detect_all_files_filter(filter);
```

Comment:
Notice how these four lines are the same for the open and save dialog? Later, you could extra another helper into an anonymous namespace that accepts these arguments and just returns an initialized `FileFilter` instance.